### PR TITLE
Added en passant method

### PR DESCRIPTION
**Pull request by jy300x2**
-Added method to track whether pawn moved one or two rows from starting position
-Added method to enable pawn to capture opposing pawn through en passant
-Added en passant field to pieces table
-Added tests in pawn_spec.rb file 

Code still needs a lot of work. I'll keep refactoring and write up more/better tests.

### CODE REVIEW

**Comment by jy300x2** [comment]
I went ahead and did some refactoring, although there's still room for improvement. Hoping I'm moving in the right track. Still need to write more/better tests. I'm open to feedback as I'm sure there's a more efficient way of checking and capturing through en passant.

File: app/models/pieces/pawn.rb [hunk 1]
```diff
@@ -11,18 +11,105 @@ def valid_move?(destination_row, destination_column)
...
+  # through en passant. I also added another field to the pieces table so that the game
+  # can keep track the pawn's en passant status. Not sure if this is the best way to
+  # track en passant status
+  def change_enpassant_status
+    case color
+    when 'black'
+      update_attributes(en_passant: false) if row_coordinate == 5
+    when 'white'
+      update_attributes(en_passant: false) if row_coordinate == 2
+    end
```
Comment (mscoutermarsh) [inline app/models/pieces/pawn.rb]: I forget, what happens if the pawn moves up 2 spaces in the first move? It should change to `false` the too, right?

**Comment by mscoutermarsh** [comment]
Looks great. Easy to read / looks good to me.

For hooking this up in the controller. What's your plan on that? Maybe add into `move_to`?

**Comment by jy300x2** [comment]
Josh noticed that the bishop piece didn't have an obstructed method in it so I went ahead and added that and a few tests.

**Comment by jy300x2** [comment]
@mscoutermarsh I agree, putting it in move_to? should work

**Pull request merged**